### PR TITLE
fix: keyboard shortcut formatting error fix. Fixes #416 

### LINKED ISSUE #416: [Bug] Keyboard shorcuts for bold, underline and italics don't work without selecting text.
### Bug Description

Keyboard shortcuts for bold, underline and italics (cmd+b, cmd+u, cmd+i) works fine when some text is selected. It applies the formatting on the selected text and continues the same formatting unless specified otherwise. But you cannot get it to work without selecting any text.

just pressing ctrl + b does nothing. We need to select some text for it to work.

### How to Reproduce

1. Open Appflowy (the editor)
2. type something anywhere
3. select some text
4. press ```cmd + b``` 
5. the selected text becomes bold
6. The next thing you type is still bold
7. You cannot disable this by pressing ``` cmd + b``` again
8. You need to select text again and press ```cmd+b``` to do so.

### Expected Behavior

We should be able to toggle these three features without selecting the text

### Operating System

Ubuntu 22.04.2 LTS

### AppFlowy Editor Version(s)

1.2.3 (Appflowy 0.3.0)

### Screenshots

Screenshot from 2023-08-25 17-32-54
Screenshot from 2023-08-25 17-33-05
Screenshot from 2023-08-25 17-33-47
Screenshot from 2023-08-25 17-34-41
Screenshot from 2023-08-25 17-35-06
Screenshot from 2023-08-25 17-35-40

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Chandan-CV**
**This PR fixes the issue #416**

Steps I took to fix the issue:
 1. Having a selection is mandatory for formatting. ```selection.lenght > 0```
 2. Hence, when a keyboard shortcut is pressed, a **white space** is inserted. and the formatting is carried out on the whitespace
 3. from then, any text entered after the shortcut will have the same formatting options.

demo:

  
[Screencast from 08-09-23 01:22:34 AM IST.webm](https://github.com/AppFlowy-IO/appflowy-editor/assets/37525954/04e00a1b-a8ab-483b-a1f3-d3c27b97049b)

### CODE REVIEW

**Comment by Chandan-CV** [comment]
@Xazin, your inputs here will be greatly appreciated :)

Thank you.

**Comment by Xazin** [comment]
Hey @Chandan-CV - Sorry about the delay, I'll take a look at this the coming weekend and test it out.

The solution at first glance looks straightforward :+1:

**Comment by Chandan-CV** [comment]
Hi @Xazin! any updates on the PR?

File: lib/src/editor/command/text_commands.dart [hunk 1]
```diff
@@ -173,7 +173,20 @@ extension TextTransforms on EditorState {
...
+    if (selection.end.offset == 0) {
+      return false;
+    }
+    return getTextInSelection(
+          selection.copyWith(
+            start: selection.end.copyWith(offset: selection.end.offset - 1),
+            end: selection.end.copyWith(offset: selection.end.offset),
+          ),
+        ).first ==
+        " ";
```
Comment (Xazin) [inline lib/src/editor/command/text_commands.dart:186]: Let's abstract this to a const eg. `const whiteSpaceCharacter = " ";`

Comment (Chandan-CV) [reply lib/src/editor/command/text_commands.dart:186]: Sure, I'm on it :)

Comment (Chandan-CV) [reply lib/src/editor/command/text_commands.dart]: @Xazin, I refactored the code accordingly. Does it look right now?

Thank you!

**Comment by LucasXu0** [comment]
@Chandan-CV Thank you for your contribution. I have implemented this [feature](https://github.com/AppFlowy-IO/appflowy-editor/pull/528) by saving the toggled style.

**Comment by Xazin** [comment]
We have implemented a version of this :+1:

**Comment by Chandan-CV** [comment]
That's a really clean implementation of the feature! Thank you for showing me how it's done :)

**Pull request closed without merging**